tools/cmaketobzl/cmaketobzl.go

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/kythe/llvmbzlgen/cmakelib/ast"
+	"github.com/kythe/llvmbzlgen/cmakelib/bindings"
+	"github.com/kythe/llvmbzlgen/writer"
+)
+
+// blockCounter counts active blocks of the given name for matching
+// paired CMake commenads, e.g. if/endif
+type blockCounter struct {
+	begin string
+	end   string
+	count int
+}
+
+// newCounter returns a new blockCounter instance which counts
+// blocks delimited by begin and "end" + begin.
+func newCounter(begin string) *blockCounter {
+	return &blockCounter{begin, "end" + begin, 0}
+}
+
+// Count increments the internal counter if text matches the begin delimiter,
+// decrements it if it matches the end delimiter and returns true if text
+// matched a delimiter or the current count is greater than zero.
+func (bc *blockCounter) Count(text string) bool {
+	matched := true
+	if text == bc.begin {
+		bc.count += 1
+	} else if text == bc.end {
+		bc.count -= 1
+	} else {
+		matched = false
+	}
+	return matched || bc.count > 0
+}
+
+type eval struct {
+	p *ast.Parser
+	o options
+
+	w    *writer.StarlarkWriter
+	v    *bindings.Mapping
+	path []string
+}
+
+type options struct {
+	macroName   string
+	shouldPrint func(string) bool
+	shouldAdd   func(string) bool
+	excludePath func(string) bool
+}
+
+// Option is a configuration option for the CMake evaluator.
+type Option func(*eval)
+
+// PrintCommands configures the evaluator to print commands on the StarlarkWriter for which the supplied predicate returns true.
+func PrintCommands(p func(string) bool) Option {
+	return func(e *eval) {
```

Comment:
What you wrote is fine, but these could be compactified a bit:

```go
return func(e *eval) { e.o.shouldPrint = p }
```

if you like.

Reply:
Done.